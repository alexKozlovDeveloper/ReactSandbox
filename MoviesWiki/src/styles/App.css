h1 {
    color: #27aedb;
    text-align: center;
}

.body {
    background-color: dimgray;
}

.font {
    font-family: 'Graduate', cursive;
}