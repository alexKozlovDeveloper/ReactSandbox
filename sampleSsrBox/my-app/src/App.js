import React, { Component } from 'react';


import { Provider } from 'react-redux'
import { createStore } from 'redux'
import rootReducer from './reducers'

import 'isomorphic-fetch';
import 'babel-polyfill';
import { hot } from 'react-hot-loader';

import Home from './Pages/Home';
import About from './Pages/About';
import NotFound from './Pages/NotFound';


//import { StaticRouter as Router, Route, Switch, Link } from 'react-router-dom'
import { Route, Switch, Link } from 'react-router-dom'


//const store = createStore(rootReducer)
//

const App = ({ Router, location, context, store }) => (
<Provider store={store}>
  <Router location={location} context={context}>
    <div className="App">
      <header className="App-header">
        <div>
          <ul>
            <li>
              <Link to="/">Home</Link>
            </li>
            <li>
              <Link to="/about">About</Link>
            </li>
            <li>
              <Link to="/invalidLinkPath">404</Link>
            </li>
          </ul>
        </div>
        <div>
          <Switch>
            <Route exact path="/" component={Home} />
            <Route path="/about" component={About} />
            <Route path="*" component={NotFound} />
          </Switch>
        </div>
      </header>
    </div>
  </Router>
</Provider>
)

export default hot(module)(App);
