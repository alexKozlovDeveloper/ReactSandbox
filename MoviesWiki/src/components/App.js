import React, { Component } from "react";

import Header from "./Header";
import Footer from "./Footer";
import Home from "./Pages/Home";
import MovieView from "./Pages/MovieView";

import ErrorBoundary from "./Error/ErrorBoundary";

import styles from "../styles/App.css"

import { BrowserRouter as Router, Route } from 'react-router-dom'

class App extends Component {
    constructor(props) {
        super(props);

        this.state = {
            headerConfig: {
                title: "Find Your Movies",
                searchConfig: {
                    placeHolder: "The Godfather",
                    buttonText: "Search",
                    searchFilter: {
                        title: "Search by",
                        options: [{
                            title: "Title",
                            value: "title"
                        }, {
                            title: "Genre",
                            value: "genres"
                        }]
                    }
                }
            },
            footerConfig: {
                title: "Copyright Â© 2019 AkTest"
            },
            resultsBodyConfig: {
                itemsPerRow: 4,
                resultSortConfig: {
                    title: "Sort by",
                    options: [{
                        title: "Release date",
                        value: "release_date"
                    }, {
                        title: "Rating",
                        value: "vote_average"
                    }]
                },
                resultsCountConfig: {
                    title: "movies found"
                }
            }
        };
    }

    render() {
        return <div className={styles.font}>
            <Router>
                <ErrorBoundary>                    
                    <Header config={this.state.headerConfig} />
                    <Route path="/view" component={MovieView} />
                    <Route path="/" render={(props) => <Home {...props} resultsBodyConfig={this.state.resultsBodyConfig} />}/>
                    <Footer config={this.state.footerConfig} />
                </ErrorBoundary>
            </Router>
        </div>;
    }
}

export default App;