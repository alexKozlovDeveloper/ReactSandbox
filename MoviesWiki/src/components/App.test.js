import React from 'react';
import App from './App';
import renderer from 'react-test-renderer';

test('My test Func', () => {
  const component = renderer.create(
    <App />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  const mouseArea = tree.children[1];

  //console.log(' TREE !!! ' + JSON.stringify(mouseArea));

  mouseArea.props.onMouseEnter();

  tree = component.toJSON();
  expect(tree).toMatchSnapshot();


  mouseArea.props.onMouseLeave();

  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});