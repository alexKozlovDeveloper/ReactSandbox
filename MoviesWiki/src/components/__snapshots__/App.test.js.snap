// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App rendering test 1`] = `
<div
  className="font"
>
  <ErrorBoundary>
    <Header
      config={
        Object {
          "searchConfig": Object {
            "buttonText": "Search",
            "placeHolder": "The Godfather",
            "searchFilter": Object {
              "options": Array [
                Object {
                  "title": "Title",
                  "value": "title",
                },
                Object {
                  "title": "Genre",
                  "value": "genres",
                },
              ],
              "selectedIndex": "0",
              "title": "Search by",
            },
          },
          "title": "Find Your Movies",
        }
      }
    />
    <Connect(ItemDetails) />
    <Connect(ResultsBody)
      config={
        Object {
          "itemsPerRow": 4,
          "resultSortConfig": Object {
            "options": Array [
              Object {
                "title": "Release date",
                "value": "release_date",
              },
              Object {
                "title": "Rating",
                "value": "vote_average",
              },
            ],
            "selectedIndex": "1",
            "title": "Sort by",
          },
          "resultsCountConfig": Object {
            "title": "movies found",
          },
        }
      }
    />
    <Footer
      config={
        Object {
          "title": "Copyright Â© 2019 AkTest",
        }
      }
    />
  </ErrorBoundary>
</div>
`;
